expansion/src/main/java/no/unit/nva/expansion/model/ExpandedParentPublication.java

```diff
@@ -65,13 +63,20 @@ private List<InputStream> getInputStreams(String publicationJsonString) {
 
     private Collection<? extends InputStream> fetchAll(List<URI> externalReferences) {
         return externalReferences.stream()
+                   .filter(this::isNotBlankUri)
                    .map(this::fetch)
-                   .flatMap(Optional::stream)
                    .map(IoUtils::stringToStream)
-                   .collect(Collectors.toList());
+                   .toList();
     }
 
-    private Optional<String> fetch(URI externalReference) {
-        return uriRetriever.getRawContent(externalReference, APPLICATION_JSON_LD.toString());
+    private boolean isNotBlankUri(URI uri) {
+        return StringUtils.isNotBlank(uri.toString());
+    }
+
+    private String fetch(URI externalReference) {
+        return uriRetriever.fetchResponse(externalReference, APPLICATION_JSON_LD.toString())
+                   .filter(i -> i.statusCode() / 100 == 2)
```

Comment:
Rename `i` to `response`?
Also: I don't think we want to through on 404s and 301s

Reply:
We will look at this in a following PR. The tests will be the majority of the code.